File: python/leetcode/349. Intersection of Two Arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        which = []
        other = []
        ans = []
        d = {}
        if len(nums1) > len(nums2):
            which = nums2
            other = nums1
        else:
            which = nums1
            other = nums2

        for i in range(len(which)):
            d[which[i]] = 0
        for i in range(len(other)):
            if other[i] in d:
                ans.append(other[i])
                del d[other[i]]
        return ans

File: c++/leetcode/349. Intersection of Two Arrays.cpp
class Solution
{
public:
  vector<int> intersection(vector<int> &nums1, vector<int> &nums2)
  {
    vector<int> which, other, ans;
    if (nums1.size() > nums2.size())
    {
      which = nums2;
      other = nums1;
    }
    else
    {
      which = nums1;
      other = nums2;
    }

    set<int> st;
    for (int i = 0; i < which.size(); i++)
    {
      st.insert(which[i]);
    }

    for (int i = 0; i < other.size(); i++)
    {
      if (st.count(other[i]))
      {
        ans.push_back(other[i]);
        st.erase(other[i]);
      }
    }

    return ans;
  }
};